src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -304,6 +314,16 @@ DataReaderImpl::~DataReaderImpl()
 
 bool DataReaderImpl::can_be_deleted() const
 {
+    {
+        std::lock_guard<std::recursive_mutex> _(get_conditions_mutex());
```

Comment:
Acknowledged. To prevent any issues on notification the `get_conditions_mutex()` must be taken before the `reader_->getMutex()` one (doxygen on the mutex has been updated).
Nevertheless the `ReturnCode_t SubscriberImpl::delete_datareader(const DataReader* reader)` will not take the mutex between `DataReaderImpl::can_be_deleted()` check and the actual deletion. This happens too for the loans, that is,